src/Modal.php

```diff
@@ -29,176 +31,251 @@
  */
 final class Modal extends Widget
 {
-    public const SIZE_SMALL = 'is-small';
-    public const SIZE_MEDIUM = 'is-medium';
-    public const SIZE_LARGE = 'is-large';
-    private const SIZE_ALL = [
-        self::SIZE_SMALL,
-        self::SIZE_MEDIUM,
-        self::SIZE_LARGE,
-    ];
-
-    public const COLOR_PRIMARY = 'is-primary';
-    public const COLOR_LINK = 'is-link';
-    public const COLOR_INFO = 'is-info';
-    public const COLOR_SUCCESS = 'is-success';
-    public const COLOR_WARNING = 'is-warning';
-    public const COLOR_DANGER = 'is-danger';
-    private const COLOR_ALL = [
-        self::COLOR_PRIMARY,
-        self::COLOR_LINK,
-        self::COLOR_INFO,
-        self::COLOR_SUCCESS,
-        self::COLOR_WARNING,
-        self::COLOR_DANGER,
-    ];
-
-    private array $options = [];
-    private array $contentOptions = [];
-    private array $closeButtonOptions = [];
+    private const COLORS = ['is-primary', 'is-link', 'is-info', 'is-success', 'is-warning', 'is-danger', 'is-dark'];
+    private const SIZES = ['is-small', 'is-medium', 'is-large'];
+    private array $attributes = [];
+    private string $autoIdPrefix = 'w';
+    private array $closeButtonAttributes = [];
+    private string $closeButtonClass = 'modal-close';
     private string $closeButtonSize = '';
-    private bool $withoutCloseButton = true;
+    private array $contentAttributes = [];
+    private string $modalBackgroundClass = 'modal-background';
+    private string $modalButtonClass = 'button modal-button';
+    private string $modalClass = 'modal';
+    private string $modalContentClass = 'modal-content';
+    private array $toggleButtonAttributes = [];
     private string $toggleButtonLabel = 'Toggle button';
     private string $toggleButtonSize = '';
     private string $toggleButtonColor = '';
-    private array $toggleButtonOptions = [];
+    private bool $withoutCloseButton = true;
     private bool $withoutToggleButton = true;
 
     /**
-     * Returns a new instance with the specified main container options.
+     * The HTML attributes. The following special options are recognized.
```

Comment:
There are no special options info so

```suggestion
     * The HTML attributes.
```

Reply:
done.